### PR TITLE
change download path for geoip plugin

### PR DESCRIPTION
**Pull request by kaisecheng**
Currently, the plugin downloads database and creates metadata under logstash-filter-geoip/vendor which has permission issue in docker and dep/rmps
This PR changes the working directory to `path.data`

There are some follow-up PRs for geoip, and the CI is highly coupling to the plugin. To simply the test/CI process, the removal of `vendor_path` in DatabaseManager constructor will be in another PR

Fixed: https://github.com/elastic/logstash/issues/12858

### CODE REVIEW

File: x-pack/lib/filters/geoip/database_manager.rb [hunk 1]
```diff
@@ -26,12 +26,14 @@
 #   while `offline` is for static database path provided by users
 
 module LogStash module Filters module Geoip class DatabaseManager
+  extend LogStash::Filters::Geoip::Util
   include LogStash::Util::Loggable
   include LogStash::Filters::Geoip::Util
 
+  #TODO remove vendor_path
   def initialize(geoip, database_path, database_type, vendor_path)
```
Comment (kaisecheng) [inline x-pack/lib/filters/geoip/database_manager.rb:34]: `vendor_path` will be removed in another PR

File: x-pack/lib/filters/geoip/database_manager.rb [hunk 2]
```diff
@@ -61,6 +63,15 @@ def initialize(geoip, database_path, database_type, vendor_path)
 
   public
 
+  # create data dir, path.data, for geoip if it doesn't exist
+  # copy CC databases to data dir
+  def self.prepare_cc_db
+    FileUtils::mkdir_p(get_data_dir)
+    FileUtils.cp_r(::Dir.glob(::File.join(
+      LogStash::Environment::LOGSTASH_HOME, "vendor", "**", "{GeoLite2-ASN,GeoLite2-City}.mmdb")),
+      get_data_dir) if !::File.exist?(get_file_path(CITY_DB_NAME)) || !::File.exist?(get_file_path(ASN_DB_NAME))
```
Comment (andsel) [inline x-pack/lib/filters/geoip/database_manager.rb:72]: Probably a more readable version could be with not inlined condition:
```suggestion
   unless ::File.exist?(get_file_path(CITY_DB_NAME)) && ::File.exist?(get_file_path(ASN_DB_NAME))
     db_source_dir = ::Dir.glob(::File.join(
           LogStash::Environment::LOGSTASH_HOME, "vendor", "**", "{GeoLite2-ASN,GeoLite2-City}.mmdb"))
     FileUtils.cp_r(,db_source_dir, get_data_dir)
   end   
```

File: x-pack/lib/filters/geoip/database_manager.rb [hunk 3]
```diff
@@ -26,12 +26,14 @@
 #   while `offline` is for static database path provided by users
 
 module LogStash module Filters module Geoip class DatabaseManager
```
Comment (andsel) [inline x-pack/lib/filters/geoip/database_manager.rb:28]: In a sequence of modules in one line probably is best to separate class and module chain, to have structure like:
```ruby
module LogStash module Filters module Geoip
  class DatabaseManager
  ...
  end
end; end; end
```

or go with: 
```ruby
class LogStash::Filters::Geoip::DatabaseManager
...
end
```

**Review by andsel — APPROVED** [review]
LGTM, I've left a couple of comments on minor stylistic points
